angr/analyses/reaching_definitions/dep_graph.py

```diff
@@ -45,41 +45,41 @@ def add_node(self, node):
 
         self._graph.add_node(node)
 
-    def add_edge(self, source, destination, **labels):
+    def add_edge(self, source: Definition, destination: Definition, **labels) -> None:
         """
         The edge to add to the definition-use graph. Will create nodes that are not yet present.
 
-        :param Definition source: The "source" definition, used by the "destination".
-        :param Definition destination: The "destination" definition, using the variable defined by "source".
+        :param source: The "source" definition, used by the "destination".
+        :param destination: The "destination" definition, using the variable defined by "source".
         :param labels: Optional keyword arguments to represent edge labels.
         """
         if not _is_definition(source) and not _is_definition(destination):
             raise TypeError("In a DepGraph, edges need to be between <%s>s." % Definition.__name__)
 
         self._graph.add_edge(source, destination, **labels)
 
-    def transitive_closure(self, definition):
+    def transitive_closure(self, definition: Definition) -> networkx.DiGraph:
         """
         Compute the "transitive closure" of a given definition.
         Obtained by transitively aggregating the ancestors of this definition in the graph.
 
-        Note: Each definition is memoized to avoid any kind of recomputation accross the lifetime of this object.
+        Note: Each definition is memoized to avoid any kind of recomputation across the lifetime of this object.
 
-        :param Definition definition: The <Definition> to return the top-level ancestors for.
-        :return List[Definition]: The list of top-level definitions flowing into the <node>.
+        :param definition:  The Definition to get transitive closure for.
+        :return:            A graph of the transitive closure of the given definition.
         """
 
-        def _transitive_closure(definition, graph, result=None):
-            if definition in self._transitive_closures.keys():
-                return self._transitive_closures[definition]
+        def _transitive_closure(def_: Definition, graph: networkx.DiGraph, result: networkx.DiGraph):
```

Comment:
`def_` :sob:

Reply:
Because you don't want to shadow the `definition` in the outer scope, and `def` is a keyword.